copier/main.py

```diff
@@ -234,7 +238,7 @@ def _cleanup(self) -> None:
         for method in self._cleanup_hooks:
             method()
 
-    def _check_unsafe(self, mode: Literal["copy", "update"]) -> None:
+    def _check_unsafe(self, mode: Operation) -> None:
```

Comment:
No need to pass ˋmodeˋ if it is in ˋself.operationˋ, right?

Reply:
Initially I did it like that, but noticed that this would cause a behavior change (at least in theory):

During `_apply_update`, `self.operation` is `update`, but it calls on `run_copy` several times, which would pass `copy` to `_check_unsafe` before this patch.

Reply:
Yes. However, that is correct. You will notice that there are calls to `replace`. In those calls, you can replace some configuration for the sub-worker that is created. Could you please try doing it that way?

Reply:
Sorry, I'm not sure I'm following. First, let me sum up:

* Introducing `_copier_conf.operation` means we have an attribute on the worker representing the current high-level (user-requested) operation.
* You're proposing to use this reference for `_check_unsafe` instead of the parameter.
* I noted that doing this will change how `_check_unsafe` behaves during the individual copy operations that run during an update, where the high-level operation is `update`, but the low-level one is `copy`.

I'm already using `replace` for overriding the operation during `update`. Are you saying the high-level operation during the individual copy operations should be `copy`? Because that would mean `_copier_conf.operation` is *always* `copy`, i.e. defeat the purpose of this feature.